### PR TITLE
[RFC] Remove 'swapsync'

### PR DESCRIPTION
**Pull request by sethjackson**
See https://github.com/neovim/neovim/pull/3911#discussion_r48676049

This removes the `swapsync` option and unconditionally `fsync()`s.

cc @justinmk

### CODE REVIEW

File: src/nvim/memfile.c [hunk 1]
```diff
@@ -459,11 +459,9 @@ int mf_sync(memfile_T *mfp, int flags)
   if (hp == NULL || status == FAIL)
     mfp->mf_dirty = false;
 
-  if ((flags & MFS_FLUSH) && *p_sws != NUL) {
-    if (STRCMP(p_sws, "fsync") == 0) {
-      if (os_fsync(mfp->mf_fd)) {
-        status = FAIL;
-      }
+  if ((flags & MFS_FLUSH)) {
```
Comment (justinmk) [inline src/nvim/memfile.c]: extra parens

File: runtime/doc/recover.txt [hunk 2]
```diff
@@ -92,10 +92,6 @@ changed, not when you only moved around.  The reason why it is not kept up to
 date all the time is that this would slow down normal work too much.  You can
 change the 200 character count with the 'updatecount' option.  You can set
 the time with the 'updatetime' option.  The time is given in milliseconds.
-After writing to the swap file Vim syncs the file to disk.  This takes some
```
Comment (justinmk) [inline runtime/doc/recover.txt]: This sentence should stay don't you think? Because we always fsync now.

File: runtime/doc/options.txt [hunk 3]
```diff
@@ -6204,25 +6203,13 @@ A jump table for the options with a short description can be found at |Q_op|.
...
-			global
-	When this option is not empty a swap file is synced to disk after
-	writing to it.  This takes some time, especially on busy Unix systems.
-	When this option is empty parts of the swap file may be in memory and
-	not written to disk.  When the system crashes you may lose more work.
-	On Unix the system does a sync now and then without Vim asking for it,
-	so the disadvantage of setting this option off is small.  On some
-	systems the swap file will not be written at all. 
-	The 'fsync' option is used for the actual file.
-
```
Comment (Pyrohh) [inline runtime/doc/options.txt]: Should probably keep a note here given what's been done with the other removed options:
```
			*'compatible'* *'cp'* *'nocompatible'* *'nocp'*
'compatible' 'cp'	Removed. |vim-differences| {Nvim}
```

Comment (sethjackson) [inline runtime/doc/options.txt]: :+1: oh right. Whoops!

File: runtime/doc/recover.txt (hunk 2 above)
Comment (sethjackson) [inline runtime/doc/recover.txt]: Yes wasn't paying attention. Thanks!

Comment (Pyrohh) [inline runtime/doc/recover.txt]: :+1:

File: runtime/doc/options.txt (hunk 3 above)
Comment (Pyrohh) [inline runtime/doc/options.txt]: oh, and while you're at it, please remember to mention it in `vim_diff.txt`.

**Comment by sethjackson** [comment]
Wow seriously I **just** rebased and it's out of date. ROFL

**Comment by justinmk** [comment]
Rebase should not be needed unless there is a conflict. Are you getting conflicts?

**Comment by sethjackson** [comment]
Nope. Just GH complaining that my branch is out of date. I suppose you can merge manually but pushing buttons is cool too. :)

**Comment by Pyrohh** [comment]
Not sure if @justinmk can, but I definitely can't as we use the protected branch feature; don't get me wrong, as I think the benefits outweigh the downsides... assuming that this complaining is not related to protected branch stuff.

**Comment by Pyrohh** [comment]
Just realized, but wouldn't it be better for a dummy option to remain, like what's done for `ttyfast`? This is what I mean (from `vim_diff.txt`):
```
==============================================================================
6. Removed features					 *nvim-features-removed*

These features are in Vim, but have been intentionally removed from Nvim.

Vi-compatible mode:
  ":set nocompatible" is ignored
  ":set compatible" is an error

Ed-compatible mode:
  ":set noedcompatible" is ignored
  ":set edcompatible" is an error

'ttyfast':
  ":set ttyfast" is ignored
  ":set nottyfast" is an error

```

**Comment by sethjackson** [comment]
Hmm. I'll have to look at those commits so that `:set swapsync` is ignored since I don't know how to do that right off. Those features are booleans though right? Because `:set noswapsync` doesn't exist AFAIK.

**Comment by justinmk** [comment]
Sure.

I will disable the branch protection temporarily to see if it is the cause of that issue.

**Comment by sethjackson** [comment]
@Pyrohh those all appear to be boolean options. The other stuff removed just says `Unknown option: <option>` when trying to do a `:set`. I think in this case removing it fully should be ok since `swapsync` is a non-boolean option?

**Comment by justinmk** [comment]
Forget it then. Wait until someone complains. I think very very few people would set this to something other than the default.

**Comment by Pyrohh** [comment]
> Forget it then.

:+1: Anyways this LGTM.

File: src/nvim/memfile.c (hunk 1 above)
Comment (sethjackson) [inline src/nvim/memfile.c]: I think these superfluous parenthesis can be removed right?

Comment (Pyrohh) [inline src/nvim/memfile.c]: I think so, it looks like they were only needed when the `&&` was there).

**Comment by justinmk** [comment]
> I will disable the branch protection temporarily to see if it is the cause of that issue.

Done. LMK if the "branch out of date" thing continues.

**Comment by Pyrohh** [comment]
> Done. LMK if the "branch out of date" thing continues.

Thanks.

Oh, and thanks @sethjackson.

**Pull request merged**